File: packages/flyto-indexer/tests/fixtures/vuln/weak_crypto.js
// Fixtures for weak_crypto (JS) + TLS rules.
const crypto = require('crypto')
const bcrypt = require('bcrypt')

function weakMd5(data) {
  return crypto.createHash('md5').update(data).digest('hex') // WEAK_HASH_MD5
}

function weakSha1(data) {
  return crypto.createHash('sha1').update(data).digest('hex') // WEAK_HASH_SHA1
}

function insecureToken() {
  return Math.random().toString(36) // INSECURE_RANDOM_MATHS
}

function weakAesEcb(key, plaintext) {
  return crypto.createCipheriv('aes-128-ecb', key, null) // WEAK_AES_ECB
}

function weakBcrypt(password) {
  return bcrypt.hashSync(password, 8) // BCRYPT_LOW_ROUNDS
}

// TLS verification off
const https = require('https')
const agent = new https.Agent({ rejectUnauthorized: false }) // TLS_VERIFY_DISABLED_JS

File: packages/flyto-indexer/tests/fixtures/vuln/weak_crypto.py
# Fixtures for weak_crypto rules — every line here SHOULD fire a rule.
# Keep each vuln on its own line + no fixes so counts are predictable.

import hashlib
import random
import secrets
from Crypto.Cipher import AES
from Crypto.Protocol.KDF import PBKDF2


def weak_md5(data):
    return hashlib.md5(data).hexdigest()  # WEAK_HASH_MD5


def weak_sha1(data):
    return hashlib.sha1(data).hexdigest()  # WEAK_HASH_SHA1


def insecure_random_token():
    return str(random.random())  # INSECURE_RANDOM_PY


def insecure_random_choice(items):
    return random.choice(items)  # INSECURE_RANDOM_PY


def bad_aes_ecb(key, plaintext):
    cipher = AES.new(key, AES.MODE_ECB)  # WEAK_AES_ECB
    return cipher.encrypt(plaintext)


def hardcoded_iv(key, plaintext):
    iv = bytes(16)  # HARDCODED_IV_ZERO
    cipher = AES.new(key, AES.MODE_CBC, iv)
    return cipher.encrypt(plaintext)


def weak_kdf(password, salt):
    return PBKDF2(password, salt, 32, 1000, 'md5')  # WEAK_KDF_MD5
